From f1367ef9467416b67ab7c450555893e082d3fca6 Mon Sep 17 00:00:00 2001
From: Igor <igor@pbne.mygbiz.com>
Date: Fri, 30 Oct 2015 14:48:14 +0100
Subject: [PATCH 44/63] split: use separate apt sources.list.d/ file for
 keeping local repo, to avoid network errors with apt-get update

---
 revsw-opensource-common.mk | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/revsw-opensource-common.mk b/revsw-opensource-common.mk
index a436c78..c9ea92a 100644
--- a/revsw-opensource-common.mk
+++ b/revsw-opensource-common.mk
@@ -7,6 +7,7 @@ DIR=$(shell pwd)
 DOCKER_BIN=docker
 # set it to -ti when doing non-Jenkins build
 INTERACTIVE?=
+APT_REPO=/etc/apt/sources.list.d/revsw-varnish-on-ci.list
 DOCKER=docker run --rm $(INTERACTIVE) -v $(DIR):/work -w /work -e IN_DOCKER=true $(BUILDER)
 #WRAPPER=$(DOCKER)
 WRAPPER=sudo
@@ -53,11 +54,11 @@ clean:
 
 check-varnish-repo:
 	@mkdir -p $(DEBS) && cd $(DEBS) && dpkg-scanpackages . > Packages && rm -f *.bz2 && bzip2 -kf Packages
-	@if [ $(shell grep "$(DEBS)" /etc/apt/sources.list 2>/dev/null | wc -l) -lt 1 ]; then \
+	@if [ $(shell grep "$(DEBS)" $(APT_REPO) 2>/dev/null | wc -l) -lt 1 ]; then \
 		echo "Updating repository list to point to $(DEBS)"; \
-		echo "deb [trusted=yes] file://$(DEBS) /" >> /etc/apt/sources.list; cat /etc/apt/sources.list;\
+		echo "deb [trusted=yes] file://$(DEBS) /" >> $(APT_REPO); cat $(APT_REPO);\
 	fi
-	@apt-get update;
+	@apt-get update -o Dir::Etc::sourcelist="sources.list.d/revsw-varnish-on-ci.list" -o Dir::Etc::sourceparts="-" -o APT::Get::List-Cleanup="0";
 
 
 define BUILD_IF_NEED_STATUS
-- 
2.7.4

